src/main/java/com/coveros/selenified/element/check/azzert/AssertEquals.java

```diff
@@ -0,0 +1,278 @@
+ * of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.coveros.selenified.element.check.azzert;
+
+import com.coveros.selenified.OutputFile;
+import com.coveros.selenified.element.Element;
+import com.coveros.selenified.element.check.Equals;
+
+import java.util.Arrays;
+
+import static org.testng.AssertJUnit.*;
+
+/**
+ * Equals extends Asserts to provide some additional verification capabilities.
+ * It will handle all verifications performed on the actual element. These
+ * asserts are custom to the framework, and in addition to providing easy object
+ * oriented capabilities, they take screenshots with each verification to
+ * provide additional traceability, and assist in troubleshooting and debugging
+ * failing tests. Equals checks that elements have a particular value associated
+ * to them.
+ *
+ * @author Max Saperstone
+ * @version 3.1.0
+ * @lastupdate 2/21/2019
+ */
+public class AssertEquals implements Equals {
+
+    // this will be the name of the file we write all commands out to
+    private final OutputFile file;
+
+    // this is the element that all actions will be performed on
+    private final Element element;
+
+    public AssertEquals(Element element, OutputFile file) {
+        this.element = element;
+        this.file = file;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public OutputFile getOutputFile() {
+        return file;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Element getElement() {
+        return element;
+    }
+
+
+    // ///////////////////////////////////////
+    // assessing functionality
+    // ///////////////////////////////////////
+
+    public void matches(int expectedMatches) {
+        assertEquals("Element Match Mismatch", expectedMatches, checkMatches(expectedMatches, 0, 0));
+    }
+
+    /**
+     * Verifies that the element has a css attribute with a value equal to the
+     * value provided. If the element isn't present, or the css doesn't contain
+     * the desired attribute, this will constitute a failure, same as a
+     * mismatch. This information will be logged and recorded, with a screenshot
+     * for traceability and added debugging support.
+     *
+     * @param attribute     - the css attribute to be checked
+     * @param expectedValue the expected css value of the passed attribute of the element
+     */
+    public void cssValue(String attribute, String expectedValue) {
+        String cssValue = checkCssValue(attribute, expectedValue, 0, 0);
+        if (cssValue == null & expectedValue != null) {
+            String reason = "No element found";
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "No element found" 9 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1192)